src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -10891,6 +10898,14 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
         let isFixed, rhsExpr, overallPatTy, overallExprTy =
             match rhsExpr with
             | SynExpr.Fixed (e, _) -> true, e, NewInferenceType g, overallTy
+            // { new Foo()  } is parsed as a SynExpr.ComputationExpr.(See pars.fsy `objExpr` rule).
+            // If a SynExpr.ComputationExpr body consists of a single SynExpr.New, and it's not the argument of a computation expression builder type.
+            // Then we should treat it as a SynExpr.ObjExpr and make it consistent with the other object expressions. e.g.
+            // { new Foo } -> SynExpr.ObjExpr
+            // { new Foo() } -> SynExpr.ObjExpr
+            // { New Foo with ... } -> SynExpr.ObjExpr
+            | SynExpr.ComputationExpr(false, SynExpr.New(_, targetType, expr, m), _) ->        
+                false, SynExpr.ObjExpr(targetType, Some(expr, None), None, [], [], [], m, rhsExpr.Range), overallTy, overallTy
```

Comment:
My concern here is that we don't **usually** do such rewrites, so it feels a little odd. I think that's the question @T-Gro had on triage as well.

Reply:
See [comment](https://github.com/dotnet/fsharp/pull/17388#issuecomment-2220570369) but one way of the other we will need to do some AST rewriting to improve this analysis.

Reply:
Checking for `new` inside of SynExpr.ComputationExpr` seems adhoc and that is why I’m proposing this change